WordPress/gutenberg
# Graduate data view options out of a menu to allow more design expression

This PR implements the new view config UI proposed at https://github.com/WordPress/gutenberg/issues/63872.
To keep this PR manageable and not very big we just port the existing functionality of the old dropdown menu. (Change some parameters, and hide/show fields).
Things like moving the item size picker for the grid into the popover, or sort directly on the field list will come as follow-ups.

This PR tries to implement the patterns specified in https://github.com/WordPress/gutenberg/issues/63624.
SettingsSection has container queries to make the component responsive depending on the size of the container.

## Screenshots
<img width="464" alt="Screenshot 2024-08-01 at 17 11 34" src="https://github.com/user-attachments/assets/2629a383-0956-4920-a839-48f2d07690dd">
<img width="842" alt="Screenshot 2024-08-01 at 17 12 04" src="https://github.com/user-attachments/assets/30d7ebe2-d404-4301-bfc9-62fa57ca795f">

cc: @jameskoster 


## Testing Instructions
I verified I could configure all the fields and things worked as expected.
Using the browser inspector I added the following CSS rule:
```
.dataviews-view-config {
    width: 650px !important;
}
```
I verified I got the responsive behavior shown above.
